reactor-core/src/main/java/reactor/util/retry/Retry.java

```diff
@@ -47,6 +49,12 @@
  */
 public abstract class Retry {
 
+	private final ContextView retryContext;
```

Comment:
I was wondering if we should add `retryContext` directly to `Retry` like that. Alternatively we could isolate the feature to `RetrySpec` and `RetryBackoffSpec`, injecting the `ContextView` into the `ImmutableRetrySignal` they expose to the companion publisher they create (since it is already a `.copy()`)...

That said, the idea of the feature leaks into `RetrySignal` interface anyway, so it could be argued that it must be enforced at `Retry` level.

At a minimum, `Retry` no-arg constructor should be maintained though. This adding of a protected ctor introduces a breaking change that we can avoid.

Reply:
> I was wondering if we should add retryContext directly to Retry like that. Alternatively we could isolate the feature to RetrySpec and RetryBackoffSpec, injecting the ContextView into the ImmutableRetrySignal they expose to the companion publisher they create (since it is already a .copy())...

We had that questioning with @aneveu but the issue is this: `Retry` exists so that users can bring their own, and the framework only knows about `Retry` when it creates `RetrySignals`, so the context __has to__ belong to `Retry` (and `RetrySignal`).

> At a minimum, Retry no-arg constructor should be maintained though. This adding of a protected ctor introduces a breaking change that we can avoid.

will do

Reply:
> This adding of a protected ctor introduces a breaking change that we can avoid.

BTW, wondering why japicmp did not detect that...

Reply:
Oh, but we have Retry on branch 3.3.x but not in 3.3.0.RELEASE, that is why...

Reply:
> Oh, but we have Retry on branch 3.3.x but not in 3.3.0.RELEASE, that is why...

Ah damn!

> The framework only knows about Retry when it creates RetrySignals, so the context has to belong to Retry (and RetrySignal).

Yeah I saw that... `RetryContext` is a bit different from all the other spec parameters, in the sense that it affects the `RetrySignal`, so yeah it must mean it has to live in `Retry` directly.